### PR TITLE
Fix wheel_file_re RegExp

### PR DESCRIPTION
**Pull request by AlekseyLobanov**
# Pull Request Check List

Resolves: #9769

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

- For performance POC check related issue
- All interfaces are the same
- It covers two problems but I believe that they are close enough (few characters in the single RE)

### CODE REVIEW

**Comment by dimbleby** [comment]
I have no idea why the code thinks that `dist-info` is a valid wheel extension, it probably is just legacy wrongness that should be removed.

If it has never even worked in all these years anyway, that's pretty good evidence that it is safe to remove it.

**Review by radoering — DISMISSED** [review]
I agree. Let's remove the `dist-info` part.

**Comment by AlekseyLobanov** [comment]
I removed `.dist-info` extension from RE. No problems with tests.

File: tests/utils/test_patterns.py [hunk 1]
```diff
@@ -42,6 +54,12 @@ def test_wheel_file_re(filename: str, expected: dict[str, str | None]) -> None:
     assert groups == expected
 
 
+@pytest.mark.parametrize("filename", [".dist-info"])
+def test_wheel_file_re_not_matches(filename: str) -> None:
+    match = patterns.wheel_file_re.match(filename)
+    assert match is None
+
+
```
Comment (radoering) [inline tests/utils/test_patterns.py:62]: This test is no longer of much use, is it? Thus, I think we can remove it.

Comment (AlekseyLobanov) [reply tests/utils/test_patterns.py:62]: Yes, agree. Test has been removed.

**Review by Secrus — APPROVED** [review]

**Pull request merged**